Text of all minutes files get written to database

GuyLiner opened the pull request:
- Config file to determine location of database and minutes folder
- PDF to database
- Multithreading for downloading files
- Multiproc for tesseract
#7

GuyLiner committed:
Config file and initial pdf to database functionality

A config file is now used to reference where the minutes
and database directories are. The config file is stored
in ~/.config/cobb_tracker. pdf_parse was also added
to enter the text of PDFs into the database

GuyLiner committed:
All files in the minutes/ directory get written to the database

Each file will be written to the database with the primary key.

GuyLiner committed:
Renamed imports and readded parser

GuyLiner committed:
Updated requirements.txt

GuyLiner committed:
If -m is not specified, and -p is not specified, programs exits

GuyLiner committed:
Print out error message if municipality provided is not supported

GuyLiner committed:
Added municipality as a primary key for database entry

GuyLiner committed:
Made marietta use the write_minutes_doc function

Marietta uses the write_minutes doc function in file_ops
and I removed container_name from file path. There would be extra
logic required just for marietta to be parsed differently and
entered into the database.

GuyLiner committed:
Removed message that prints when writing to disk

GuyLiner committed:
Documentation

GuyLiner committed:
Moved more file pulling functionality to file_ops

GuyLiner committed:
Moved all file ops to file_ops.py and refactoring

All operations dealing with file writing or now in
file_ops.py. Documents are now also now processed
after all of the links have been pulled, instead of
after all links in a given year are pulled.

GuyLiner committed:
Fixed return error and added more file name handling

GuyLiner committed:
threading for marietta

Marietta now uses file_ops.py to thread instead of handling it's own
file writes. Marietta also has slightly different meeting names
for a lot of the body meetings, and with this change tha also means
that each of those meetings will have their own folder under
Marietta/ instead of being under Marietta/{body}/{body_meeting_name}

GuyLiner committed:
Cobb now uses file_ops

GuyLiner committed:
Removed commented out lines

GuyLiner committed:
Moved file listing function to it's own class

GuyLiner committed:
Files are now processed in batches, and added documentation

added database_ops classes to pdf_parse and limited the amount of
processes that can be open at once to 300.

GuyLiner committed:
Removed comments

GuyLiner committed:
Documentation and more detailed error message

When the file exists, a the program will no longer print to the string 'File exists'

GuyLiner committed:
Exit if not running on *nix system

GuyLiner committed:
Imported os

GuyLiner committed:
Switched to concurrent.futures.ThreadsPoolExecutor instead of Threads() and BoundedSemaphore()

GuyLiner committed:
Removed user info from requirements.txt

GuyLiner committed:
Added config file checks

Also changed the default data directory to ~/.local/share/cobb_tracker/

GuyLiner committed:
Updated requirements.txt

GuyLiner committed:
Updated dependencies

GuyLiner committed:
Updated README.md

tyler-8 commented on src/cobb_tracker/cobb_config.py:
```diff
@@ -0,0 +1,50 @@
+import os
+from pathlib import Path
+
+class cobb_config():
```
Let's use some PEP8-like style. Classes are `CapWords`, functions & variables are `snake_case`

```python
class CobbConfig:
```

tyler-8 commented on src/cobb_tracker/cobb_config.py:
```diff
@@ -0,0 +1,50 @@
+class cobb_config():
+    def __init__(
+            self,
+            config_dir = os.path.join(Path(os.path.expanduser('~')),
```
If we're using `pathlib` we don't need `os.path`.

```python
p = pathlib.Path('parent_folder')
p.exists()
child_folder = p.joinpath("child_folder")


```

tyler-8 commented on src/cobb_tracker/cobb_config.py:
```diff
@@ -0,0 +1,50 @@
+        self.config = configparser.ConfigParser()
+
+        if not os.path.exists(self.DEFAULT_CONFIG_DIR):
+            os.mkdir(self.DEFAULT_CONFIG_DIR)
```
```python
self.DEFAULT_CONFIG_DIR.mkdir(exists_ok=True)
```

tyler-8 commented on src/cobb_tracker/municipalities/file_ops.py:
```diff
@@ -1,32 +1,85 @@
 import sys
 import os
 import requests
+import concurrent.futures
```
You can shorten later usage by doing

```python
import concurrent.futures as cf

cf.ThreadPoolExecutor()
```

tyler-8 reviewed (changes requested):
Looking good. I think we should probably consider adding the `ruff` formatter/linter to the project. It'll help with some of the formatting and keeping it consistent.

GuyLiner committed:
Class names now use CapWords convention

GuyLiner committed:
Moved from os.path to pathlib.Path

GuyLiner committed:
changed concurrent.futures import to use alias

GuyLiner committed:
Fixing redunant Path() statements

GuyLiner commented:
> Looking good. I think we should probably consider adding the `ruff` formatter/linter to the project. It'll help with some of the formatting and keeping it consistent.

Yeah, I definitely agree

GuyLiner committed:
Removed civic-scraper from requirements.txt

civic-scraper was never a dependency for us
and was never used. This accidentally got added
because I did a pip freeze > requirements.txt
after testing civic-scraper.

tyler-8 reviewed (approved):
LGTM!
